tests/matrix_free/point_evaluation_17.cc

```diff
@@ -0,0 +1,165 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2021 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+// Check precomputed mapping information against mapping
+// information on the fly for a vector valued FE and MappingCartesian.
+// This is a modified version of point_evaluation_13.cc.
+
+#include <deal.II/base/function_lib.h>
+
+#include <deal.II/dofs/dof_handler.h>
+
+#include <deal.II/fe/fe_q.h>
+#include <deal.II/fe/fe_system.h>
+#include <deal.II/fe/mapping_cartesian.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/tria.h>
+
+#include <deal.II/lac/vector.h>
+
+#include <deal.II/matrix_free/fe_point_evaluation.h>
+
+#include <deal.II/numerics/vector_tools.h>
+
+#include <iostream>
+
+#include "../tests.h"
+
+
+
+template <int dim>
+class MyFunction : public Function<dim>
+{
+public:
+  MyFunction()
+    : Function<dim>(dim)
+  {}
+
+  double
+  value(const Point<dim> &p, const unsigned int component) const override
+  {
+    return component + p[component];
+  }
+};
+
+
+
+template <int dim>
+void
+test()
+{
+  using namespace dealii;
+  Triangulation<dim> tria;
+
+  GridGenerator::subdivided_hyper_cube(tria, 2, 0, 1);
+
+  MappingCartesian<dim> mapping;
+  deallog << "Cartesian linear mapping" << std::endl;
+
+  std::vector<Point<dim>> unit_points;
+  for (unsigned int i = 0; i < 13; ++i)
+    {
+      Point<dim> p;
+      for (unsigned int d = 0; d < dim; ++d)
+        p[d] = static_cast<double>(i) / 17. + 0.015625 * d;
+      unit_points.push_back(p);
+    }
+
+  FESystem<dim>   fe(FE_Q<dim>(2), dim);
+  DoFHandler<dim> dof_handler(tria);
+  dof_handler.distribute_dofs(fe);
+  Vector<double> vector(dof_handler.n_dofs());
+
+  // mapping info object for precomputed mapping
+  NonMatching::MappingInfo<dim, dim> mapping_info(mapping,
+                                                  update_values |
+                                                    update_gradients);
+  // The first evaluator will use the precomputed mapping
+  FEPointEvaluation<dim, dim> evaluator1(mapping_info,
+                                         fe,
+                                         update_values | update_gradients);
```

Comment:
You passing here multiple times the mapping flags. What about only passing it only to `MappingInfo` and add add a getter there?

Reply:
I currently do not store the `UpdateFlags` in `MappingInfo` but only select the correct mapping flags. If you still think its better with a getter I can implement this.

Reply:
Like this @peterrum ? https://github.com/dealii/dealii/pull/13835/commits/5e2c604e60b48638af787602ee03980cb2a4185c